app/import_functions.go

```diff
@@ -1159,64 +1161,74 @@ func (a *App) getChannelsForPosts(teams map[string]*model.Team, data []*PostImpo
 	return channels, nil
 }
 
-func (a *App) importMultiplePosts(data []*PostImportData, dryRun bool) *model.AppError {
-	if len(data) == 0 {
-		return nil
+// getPostStrID returns a string ID composed of several post fields to
+// uniquely identify a post before it's imported, so it has no ID yet
+func getPostStrID(post *model.Post) string {
+	return fmt.Sprintf("%d%s%s", post.CreateAt, post.ChannelId, post.Message)
+}
+
+func (a *App) importMultiplePostLines(lines []LineImportWorkerData, dryRun bool) (*model.AppError, int) {
+	if len(lines) == 0 {
+		return nil, 0
 	}
 
-	for _, postData := range data {
-		if err := validatePostImportData(postData, a.MaxPostSize()); err != nil {
-			return err
+	for _, line := range lines {
+		if err := validatePostImportData(line.Post, a.MaxPostSize()); err != nil {
+			return err, line.LineNumber
 		}
 	}
 
 	// If this is a Dry Run, do not continue any further.
 	if dryRun {
-		return nil
+		return nil, 0
 	}
 
 	usernames := []string{}
 	teamNames := []string{}
-	for _, postData := range data {
-		usernames = append(usernames, *postData.User)
-		if postData.FlaggedBy != nil {
-			usernames = append(usernames, *postData.FlaggedBy...)
+	postsData := []*PostImportData{}
```

Comment:
nit: We can achieve some performance improvement by pre-allocations.
```suggestion
	postsData := make([]*PostImportData, len(lines)
```

Reply:
Fixed in 16f2175